keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -460,58 +402,65 @@ message PodSandboxAttributes {
 }
 
 // PodSandboxStats provides the resource usage statistics for a pod.
+// The linux or windows field will be populated depending on the platform.
 message PodSandboxStats {
     // Information of the pod.
-    // Corresponds to PodRef in SummaryAPI
     PodSandboxAttributes attributes = 1;
-    // CPU usage gathered from the pod.
-    // Corresponds to Stats SummaryAPI CPUStats field
-    CpuUsage cpu = 2;
-    // Memory usage gathered from the pod.
-    // Corresponds to Stats SummaryAPI MemoryStats field
-    MemoryUsage memory = 3;
-    // TODO: do we want a start time field?
-    // The time at which data collection for the pod-scoped (e.g. network) stats was (re)started.
-    // int64 timestamp = 1;
-    // Stats of containers in the measured pod.
-    repeated ContainerStats containers
-    // Stats pertaining to CPU resources consumed by pod cgroup (which includes all containers' resource usage and pod overhead).
-    NetworkStats network = 4;
-    // Note the specific omission of VolumeStats and EphemeralStorage
-    // Each of these fields will be calculated Kubelet-level
-    // ProcessStats pertaining to processes.
-    ProcessStats process = 5;
+    // Stats from linux.
+    LinuxPodSandboxStats linux = 2;
+    // Stats from windows.
+    WindowsPodSandboxStats windows = 3;
 }
 
-// NetworkStats contains data about network resources.
-message NetworkStats {
-    // The time at which these stats were updated.
-    int64 timestamp = 1;
+// LinuxPodSandboxStats provides the resource usage statistics for a pod sandbox on linux.
+message LinuxPodSandboxStats {
+    // CPU usage gathered for the pod sandbox.
+    CpuUsage cpu = 1;
+    // Memory usage gathered for the pod sandbox.
+    MemoryUsage memory = 2;
+    // Network usage gathered for the pod sandbox
+    NetworkUsage network = 3;
+    // Stats pertaining to processes in the pod sandbox.
+    ProcessUsage process = 4;
+    // Stats of containers in the measured pod sandbox.
+    repeated ContainerStats containers = 5;
+}
 
-    // Stats for the default interface, if found
-    InterfaceStats default_interface = 2;
+// WindowsPodSandboxStats provides the resource usage statistics for a pod sandbox on windows
+message WindowsPodSandboxStats {
+    // TODO: Add stats relevant to windows.
+}
 
-    repeated InterfaceStats interfaces = 3;
+// NetworkUsage contains data about network resources.
+message NetworkUsage {
```

Comment:
Should this be named `NetworkStats`?

Reply:
it is mirroring the Cpu and Memory structs: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L611-L621